math/unuran/inc/TUnuranSampler.h

```diff
@@ -154,29 +170,32 @@ class TUnuranSampler : public ROOT::Math::DistSampler {
 
 protected:
 
-   //initialization for 1D distributions
+   /// Initialization for 1D distributions.
    bool DoInit1D(const char * algo);
-   //initialization for 1D discrete distributions
+   /// Initialization for 1D discrete distributions.
    bool DoInitDiscrete1D(const char * algo);
-   //initialization for multi-dim distributions
+   /// Initialization for multi-dim distributions.
    bool DoInitND(const char * algo);
 
 
 private:
 
    // private member
-   bool                              fOneDim;      // flag to indicate if the function is 1 dimension
-   bool                              fDiscrete;    // flag to indicate if the function is discrete
-   bool                              fHasMode;     // flag to indicate if a mode is set
-   bool                              fHasArea;     // flag to indicate if a area is set
-   int                               fLevel;       // debug level
-   double                            fMode;        // mode of dist
-   double                            fArea;        // area of dist
-   const ROOT::Math::IGenFunction *  fFunc1D;      // 1D function pointer
-   TUnuran *                         fUnuran;      // unuran engine class
-
-   //ClassDef(TUnuranSampler,1)  //Distribution sampler class based on UNU.RAN
-
+   bool                              fOneDim = false;      /// flag to indicate if the function is 1 dimension
+   bool                              fDiscrete = false;    /// flag to indicate if the function is discrete
+   bool                              fHasMode = false;     /// flag to indicate if a mode is set
+   bool                              fHasArea = false;     /// flag to indicate if a area is set
+   bool                              fUseLogPdf = false;   /// flag to indicate if we use the log of the PDF 
+   int fLevel;                                     /// debug level
+   double                            fMode;        /// mode of dist (1D)
+   double                            fArea;        /// area of dist
+   std::vector<double>               fNDMode;      /// mode of the multi-dim distribution
+   const ROOT::Math::IGenFunction *  fFunc1D = nullptr;      /// 1D function pointer (pdf)
```

Comment:
@lmoneta I am confused with the 1D in `fFunc1D` and the comments. Isn't it used for 1 and more dimensions? (The same for `fDPDF`

Reply:
Those function objects will be used only for the 1D case. For the multi-dim case the function pointer stared in the base class DistSampler will be used